test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -5283,6 +5283,25 @@ BOOST_AUTO_TEST_CASE(simple_throw)
 	BOOST_CHECK(callContractFunction("f(uint256)", u256(1)) == encodeArgs());
 }
 
+BOOST_AUTO_TEST_CASE(fixed_arrays_as_return_type)
+{
+	char const* sourceCode = R"(
+		contract A {
+			function f() constant returns (uint16[5] arr)
+			{
+			}
+		}
+		contract B {
+			function f()
+			{
+				var a = new A();
+				uint16[5] memory res = a.f();
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "B");
```

Comment:
Please extend the test such that
 - further memory is allocated and would overwrite `res` if done incorrectly
 - some value from `res` is actually returned back to the caller - you could just let `B.f` also return `res` again.